packages/cel/src/list.ts

```diff
@@ -0,0 +1,139 @@
+// Copyright 2024-2025 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { type ReflectList, isReflectList } from "@bufbuild/protobuf/reflect";
+import { celFromListElem } from "./proto.js";
+
+/**
+ * List is common abstraction for list types.
+ */
+export abstract class List implements Iterable<unknown> {
```

Comment:
Perhaps it would be better to prefix this with `Cel` and call it the same as the old one `CelList`

Reply:
Also, we can probably change this to an interface with an internal symbol to ensure all implementations come from us and add a function check for type.